src/relay/transforms/simplify_expr.cc

```diff
@@ -163,6 +141,69 @@ class SimplifyTranspose : public DFPatternRewrite {
     return x;
   }
 
+  String PermuteLayout(const String& layout, std::vector<int> axes) const {
+    std::string new_layout{};
+    std::string old_layout{layout};
+    for (auto axis : axes) {
+      new_layout += old_layout[axis];
+    }
+    return String(new_layout);
+  }
+
+  Optional<Expr> FoldRankChangingLayoutTrans(const Expr& data, const Call& call) const {
+    Optional<Expr> layout_trans;
+    if (auto attr = call->attrs.as<LayoutTransformAttrs>()) {
+      Layout src_layout(attr->src_layout);
+      Layout dst_layout(attr->dst_layout);
+      if (src_layout->axes.size() != dst_layout->axes.size()) {
```

Comment:
It would be better to lift this logic to the beginning of this function so that people can clearly know this function only dealing with the case that source and destination layout axis numbers are mismatching.